### PR TITLE
Global Palette Elementor Compatibility

### PR DESCRIPTION
**Pull request by patilvikasj**
### Description
Display global palette colors inside Elementor global colors list 

### Screenshots
Elementor+_+Elementor+#23+-+Google+Chrome+2021-05-26+at+4 39 40+PM

### Checklist:
- [x] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: inc/compatibility/class-astra-elementor.php [hunk 1]
```diff
@@ -244,6 +251,117 @@ private function is_elementor_editor() {
...
+		/**
+		 * Display global paltte colors on Elementor front end Page.
+		 *
+		 * @since x.x.x
+		 * @param object $response rest request response.
+		 * @param array  $handler Route handler used for the request.
+		 * @param object $request Request used to generate the response.
+		 * @return object
+		 */
+		public function display_global_colors_front_end( $response, $handler, \WP_REST_Request $request ) {
```
Comment (Nikschavan) [inline inc/compatibility/class-astra-elementor.php:301]: We are not checking the route like it is checked in `elementor_add_theme_colors()`, does this mean this method will run on all the rest api calls?

File: inc/compatibility/class-astra-elementor.php [hunk 2]
```diff
@@ -244,6 +251,117 @@ private function is_elementor_editor() {
...
+			}
+
+			$rest_id = substr( $route, strrpos( $route, '/' ) + 1 );
+
+			if ( ! in_array( $rest_id, array_keys( $slug_map ), true ) ) {
+				return $response;
+			}
+
+			$colors   = astra_get_option( 'global-color-palette' );
+			$response = new \WP_REST_Response(
```
Comment (Nikschavan) [inline inc/compatibility/class-astra-elementor.php:319]: You can `use WP_REST_Response` at the start of the file.

Comment (Nikschavan) [reply inc/compatibility/class-astra-elementor.php:319]: Same for `WP_REST_Request`

**Review by Nikschavan — COMMENTED** [review]
Added a few questions

File: inc/compatibility/class-astra-elementor.php (hunk 1 above)
Comment (patilvikasj) [reply inc/compatibility/class-astra-elementor.php]: I have added a check at the beginning of the method now.

File: inc/compatibility/class-astra-elementor.php (hunk 2 above)
Comment (patilvikasj) [reply inc/compatibility/class-astra-elementor.php]: Using `rest_ensure_response` function instead now.

**Review by Nikschavan — APPROVED** [review]

**Comment by premanshup** [comment]
@patilvikasj For Dark theme, the background color is not appearing properly for Theme related Global Color - [https://share.bsf.io/OAuQmAmo](https://share.bsf.io/OAuQmAmo), please check.

File: inc/customizer/extend-custom-controls/src/color-palette/color-palette.js [hunk 3]
```diff
@@ -75,38 +76,42 @@ const ColorPaletteComponent = (props) => {
...
 
-	const handleColorReset = ( index, color ) => {
-
+	const handleColorReset = (index, color) => {
 		let updateState = {
 			...state,
 		};
 
-		let resetValue = defaultValue.palettes[updateState.currentPalette][index];
+		let resetValue =
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/color-palette/color-palette.js:84]:
```suggestion
		const resetValue =
```

File: inc/customizer/extend-custom-controls/src/color-palette/color-palette.js [hunk 4]
```diff
@@ -75,38 +76,42 @@ const ColorPaletteComponent = (props) => {
...
 		updateState.palettes[updateState.currentPalette][index] = resetValue;
-		updateValues( updateState );
+		updateValues(updateState);
 	};
 
 	var paletteColors = (
 		<>
 			<div className="ast-single-palette-wrap">
 				{state.palettes[state.currentPalette].map((value, index) => {
+					let paletteLables = astra.customizer.globalPaletteLabels;
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/color-palette/color-palette.js:95]:
```suggestion
					const paletteLables = astra.customizer.globalPaletteLabels;
```

File: inc/customizer/extend-custom-controls/src/common/astra-color-picker-control.js [hunk 5]
```diff
@@ -140,9 +140,9 @@ class AstraColorPickerControl extends Component {
 
 		Object.entries(globalColorPalette.palette).forEach(([ index, color])=>{
 			let palettePrefix = astra.customizer.globalPaletteStylePrefix;
-
+			let paletteLables = astra.customizer.globalPaletteLabels;
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/common/astra-color-picker-control.js:143]:
```suggestion
			const paletteLables = astra.customizer.globalPaletteLabels;
```

**Review by premanshup — CHANGES_REQUESTED** [review]

**Review by vrundakansara — APPROVED** [review]

**Comment by patilvikasj** [comment]
> @patilvikasj For Dark theme, the background color is not appearing properly for Theme related Global Color - https://share.bsf.io/OAuQmAmo, please check.

@premanshup  We are now losing style for dark mode also.

**Review by premanshup — APPROVED** [review]

**Pull request merged**